### ISSUE TITLE
Add property types to make:entity and rely on type inference from Doctrine ORM 2.9

### ISSUE THREAD

**Issue by weaverryan** [symfony/maker-bundle#982]
Hi!

If you're using attributes, we currently generate code that does *not* yet include the property types. We should add those!

```diff
#[ORM\Entity(repositoryClass: VictoriousPuppyRepository::class)]
class VictoriousPuppy
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
-    private $id;
+    private ?int $id;

    #[ORM\Column(type: 'string', length: 255)]
-    private $name;
+    private ?string $name;

    #[ORM\Column(type: 'boolean')]
-    private $isCute;
+    private ?bool $isCute;

    // ...
}
```

(side note, for boolean properties, it might be cool to ask for a default value in `make:entity` so we can assign the property to that and remove the `?` nullable part before `bool`).

Additionally, starting in Doctrine ORM 2.9, if your property has a type, then the `type: 'boolean'` for the `Column` attribute can be *inferred*. So for many types, you can omit the `type: `. The final result would be:

```php
#[ORM\Entity(repositoryClass: VictoriousPuppyRepository::class)]
class VictoriousPuppy
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column()]
    private ?int $id;

    #[ORM\Column(length: 255)]
    private ?string $name;

    #[ORM\Column()]
    private ?bool $isCute;
}
```

**Comment by SimonMarx** [comment]
@karaka200

I think removing the column attributes is not a good idea.
The developer should define which property create a column in the database table and it should not generally get a property also a column.

This would work in most cases, but there may be special cases where properties are set that should not exist in the table (e.g. a "changed" property which is default false and set to true when using a setter of the entity). Entities should never contain logic, but you never know... 

@weaverryan 

Should be all properties be "nullable" by default or only properties which are allowing null within the column definition by setting the nullable option?

```sql
`name` VARCHAR(255) NOT NULL
`nullable_column` VARCHAR(255)
```

```php
#[ORM\Column(length=255)]
private string $name;

#[ORM\Column(length=255, nullable=true)]
private ?string $nullableColumn = null;

public function getName(): ?string
{
    return $this->name ?? null;
}

public function setName(string $name): self
{
    $this->name = $name;
    return $this;
} 

public function getNullableColumn(): ?string
{
    return $this->nullableColumn;
}

public function setNullableColumn(?string $nullableColumn): self
{
    $this->nullableColumn = $nullableColumn;
    return $this;
}
```

Properties would be more type-safe (cannot set a null where a null is not allowed) that way and matching the column definition.
With the null coalescing operator on "not nullable property getters" we prevent errors like `Typed property must not be accessed before initialization`.

**Comment by stof** [comment]
if you remove the attribute entirely, your property is not persisted anymore.

**Comment by seb-jean** [comment]
My PHPStan will love this 😄 .

**Comment by lsv** [comment]
If im not mistaken, symfony form when create a new $form->createForm() - or maybe before, actually sends NULL values for every property in the entity - so the form can be generated.

So at the moment, every field needs to either be nullable (PHP-wise) or have a default value like `private string $foo = '';` otherwise symfony form will throw errors around.

**Comment by jrushlow** [comment]
Implemented since v1.44.0
